File: fmi-kentrack-ticket-system/frontend/src/model/Ticket.ts
import {Board} from "./Board";
import {Comment} from "./Comment";
import {Section} from "./Section";
import {UserView} from "./UserView";

export interface Ticket {
    id: number,
    title: string,
    description: string,
    owner: UserView,
    assignee: UserView,
    comments: Comment[],
    section: Section,
    board: Board,
    createdDate: Date,
    updatedDate: Date,
}

File: fmi-kentrack-ticket-system/backend/src/main/kotlin/com/fmi/kentrack/model/Ticket.kt
package com.fmi.kentrack.model

import com.fmi.kentrack.users.User
import java.time.LocalDateTime

data class Ticket(
    val id: Int? = null,
    val title: String,
    val description: String,
    val owner: User? = null,
    val assignee: User? = null,
    val comments: List<Comment>? = null,
    val section: Section? = null,
    val board: Board? = null,
    val createdDate: LocalDateTime,
    val updatedDate: LocalDateTime,
)